internal/server/spanner/model.go

```diff
@@ -15,13 +15,15 @@
 // Model objects related to the spanner graph database.
 package spanner
 
-// Edge struct represents a single row in the Edge table.
+// Edge struct represents a single row in the Edge table, supplemented with the object name and types.
 type Edge struct {
-	SubjectID   string `spanner:"subject_id"`
-	Predicate   string `spanner:"predicate"`
-	ObjectID    string `spanner:"object_id"`
-	ObjectValue string `spanner:"object_value"`
-	Provenance  string `spanner:"provenance"`
+	SubjectID   string   `spanner:"subject_id"`
+	Predicate   string   `spanner:"predicate"`
+	ObjectID    string   `spanner:"object_id"`
+	ObjectValue string   `spanner:"object_value"`
+	Provenance  string   `spanner:"provenance"`
+	Name        string   `spanner:"name"`
```

Comment:
Thought more about this and came to a conclusion that modeling name and types with edges this way makes sense since that is how we'll always manifest them in the mixer layer.

So let's go with this approach.